src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java

```diff
@@ -38,16 +38,12 @@ protected void processParsedEnvelopeDocuments(
         List<Pdf> pdfs,
         CloudBlockBlob cloudBlockBlob
     ) {
-        if (!uploadParsedEnvelopeDocuments(envelope, pdfs)) {
-            return;
+        if (uploadParsedEnvelopeDocuments(envelope, pdfs)
+            && markAsUploaded(envelope)
+            && markAsProcessed(envelope)
+            ) {
```

Comment:
I find a big "if clause" less readable than the previous version with multiple if clauses, but again, if you care about it, go ahead.